src/main/java/dev/slimevr/vr/processor/skeleton/LegTweaks.java

```diff
@@ -870,6 +772,130 @@ private float calculateCorrectionWeight(
 				* (CORRECTION_WEIGHT_MAX - CORRECTION_WEIGHT_MIN);
 	}
 
+	// calculate the center of mass of the user for the current frame
+	// returns a vector representing the center of mass position
+	private Vector3f computeCenterOfMass() {
+		// preform a check to see if the needed data is available
+		if (
+			skeleton.headNode == null
+				|| skeleton.chestNode == null
+				|| skeleton.waistNode == null
+				|| skeleton.leftFootNode == null
+				|| skeleton.rightFootNode == null
+				|| skeleton.leftKneeNode == null
+				|| skeleton.rightKneeNode == null
+				|| skeleton.leftHipNode == null
+				|| skeleton.rightHipNode == null
+
+		) {
+			return null;
+		}
+
+		// check if arm data is available
+		boolean armsAvaliable = true;
+		if (
+			!skeleton.hasLeftArmTracker
+				|| !skeleton.hasRightArmTracker
+		) {
+			armsAvaliable = false;
+		}
+
+		Vector3f centerOfMass = new Vector3f();
+
+		// compute the center of mass of smaller body parts and then sum them up
+		// with their respective weights
+		Vector3f head = skeleton.headNode.worldTransform.getTranslation();
+		Vector3f chest = skeleton.chestNode.worldTransform.getTranslation();
+		Vector3f waist = skeleton.waistNode.worldTransform.getTranslation();
+		Vector3f leftCalf = skeleton.leftAnkleNode.worldTransform
+			.getTranslation(null)
+			.add(skeleton.leftKneeNode.worldTransform.getTranslation(null))
+			.mult(0.5f);
+		Vector3f rightCalf = skeleton.rightAnkleNode.worldTransform
+			.getTranslation(null)
+			.add(skeleton.rightKneeNode.worldTransform.getTranslation(null))
+			.mult(0.5f);
+		Vector3f leftThigh = skeleton.leftKneeNode.worldTransform
+			.getTranslation(null)
+			.add(skeleton.leftHipNode.worldTransform.getTranslation(null))
+			.mult(0.5f);
+		Vector3f rightThigh = skeleton.rightKneeNode.worldTransform
+			.getTranslation(null)
+			.add(skeleton.rightHipNode.worldTransform.getTranslation(null))
+			.mult(0.5f);
```

Comment:
maybe these calculations should be in a separate function, my copy paste detector throws warnings at me here